hi-ml-histopathology/src/histopathology/datamodules/base_module.py

```diff
@@ -14,25 +15,32 @@
 
 from health_ml.utils.bag_utils import BagDataset, multibag_collate
 from health_ml.utils.common_utils import _create_generator
+from health_ml.utils.wsi_utils import list_data_collate
 
-from histopathology.datasets.base_dataset import TilesDataset
+from histopathology.datasets.base_dataset import SlidesDataset, TilesDataset
+from histopathology.datasets.panda_dataset import PandaDataset
 from histopathology.models.transforms import LoadTilesBatchd
 
+from monai.transforms.compose import Compose
+from monai.transforms.io.dictionary import LoadImaged
+from monai.apps.pathology.transforms import TileOnGridd
+from monai.data.image_reader import WSIReader
+
 
 class CacheMode(Enum):
-    NONE = 'none'
-    MEMORY = 'memory'
-    DISK = 'disk'
+    NONE = "none"
+    MEMORY = "memory"
+    DISK = "disk"
 
 
 class CacheLocation(Enum):
-    NONE = 'none'
-    CPU = 'cpu'
-    SAME = 'same'
+    NONE = "none"
+    CPU = "cpu"
+    SAME = "same"
 
 
-class TilesDataModule(LightningDataModule):
-    """Base class to load the tiles of a dataset as train, val, test sets"""
+class HistoDataModule(LightningDataModule):
+    """Base class to load a histopathology dataset as train, val, test sets"""
 
     def __init__(self, root_path: Path, max_bag_size: int = 0, batch_size: int = 1,
                  max_bag_size_inf: int = 0,
```

Comment:
`transforms` is in the docstring but not here. Also some of the docstring might need some generalization because now they refer to tiles